src/video.cpp

```diff
@@ -640,20 +639,18 @@ namespace video {
         { "pic_timing_sei"s, 0 },
         { "max_dec_frame_buffering"s, 1 },
       },
-      // SDR-specific options
-      {
-        { "profile"s, (int) qsv::profile_h264_e::high },
```

Comment:
Do we want to remove this? I think it will cause H.264 Main profile to be used instead of High.

Reply:
Hmm, I think you're right.
ffmpeg seems to ignore `AVCodecContext` when setting codec profile and uses `QSVEncContext` for this purpose.
And we can modify `QSVEncContext` only through dictionary options.
Bringing it back is probably also going to resolve the other issues you pointed out.

Reply:
Pushed the fix.